apps/site/lib/site_web/controllers/schedule/line/helpers.ex

```diff
@@ -114,6 +115,13 @@ defmodule SiteWeb.ScheduleController.Line.Helpers do
     RouteStops.by_direction(stops[route.id], shapes, route, direction_id)
   end
 
+  @spec get_shuttle_data(Plug.Conn.t()) :: [Phoenix.HTML.safe()]
+  def get_shuttle_data(conn) do
+    now = conn.assigns[:date_time] || Util.now()
+    {:ok, data} = ShuttleDiversion.active([conn.assigns.route.id], now)
+    data
+  end
+
```

Comment:
I'm working on writing a test for this and I'm not quite sure what I want this function to do if there's no route or if it doesn't return `:ok`.

Reply:
@thecristen @digitalcora I'm looking at the definition of `ShuttleDiversion.active` and it looks like, despite what the typespec implies, it will always either return `{:ok, %ShuttleDiversion{...}}`, or it will crash due to a pattern-match fail.

If I follow, if we get to this code and there is no active diversion, the most likely explanation is that the end-user went directly to the shuttles page, without following a link there. From a UX point of view, then, the best thing to do would probably be to redirect them to the main page for the line.

Reply:
I believe the typespec is correct. If any of the `with <-` pattern-matches fail, the term that failed to match will be returned, which will be `{:error, any}` as per the typespecs of the called functions. This would occur if there was an HTTP error fetching data from the API. In this case, I think we should have an error state for the page that explains something went wrong, and it might go right if you refresh. But I'm not sure if we have a convention for such things on other pages.

Reply:
you're right @digitalcora I was reading this wrong entirely, never mind

Reply:
An error state would be good; I've added a failing test and also am unsure what convention I ought to follow here :)